Rx/v2/src/rxcpp/operators/rx-reduce.hpp

```diff
@@ -128,23 +138,39 @@ struct reduce : public operator_base<rxu::value_type_t<reduce_traits<T, SourceOp
             reduce_state_type& operator=(reduce_state_type o) RXCPP_DELETE;
         };
         auto state = std::make_shared<reduce_state_type>(initial, std::move(o));
+        auto on_completed = [state]() {
+            rxu::maybe<value_type> result;
+            try {
+                result.reset(state->result_selector(state->current));
+            } catch (const std::exception&) {
+                state->out.on_error(std::current_exception());
+                return;
+            }
+            state->out.on_next(result.get());
+            state->out.on_completed();
+        };
         state->source.subscribe(
             state->out,
         // on_next
             [state](T t) {
-                auto next = state->accumulator(state->current, t);
-                state->current = next;
+                try {
```

Comment:
This try/catch should not be needed. The subscriber will call this on_next method under protection, if it throws, it will call on_error with the exception.